### PR TITLE
Add cargo-make support

### PR DESCRIPTION
**Pull request by topecongiro**
This PR adds support for cargo-make to ease the development of rustfmt. Also, I removed .cargo/config file as it was not working as intended when we run `cargo` command in subdirectories.

### CODE REVIEW

**Comment by calebcartwright** [comment]
Good catch on the relative paths in the old cargo config :man_facepalming: 

It's surprising to see the i686 targets on Windows failing, and I don't see anything in the cargo make config that explains it

File: Makefile.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+run_task = "test-subproject"
+
+[tasks.test-lib]
+env = { "RUSTFMT_MAKE_MANIFEST_PATH" = "rustfmt-core/rustfmt-lib/Cargo.toml" }
+run_task = "test-subproject"
+
+[tasks.test-subproject]
+condition = { env_set = ["RUSTFMT_MAKE_MANIFEST_PATH"] }
+script = [
+  "cargo test --manifest-path ${RUSTFMT_MAKE_MANIFEST_PATH}",
```
Comment (calebcartwright) [inline Makefile.toml:38]: seems the Windows jobs aren't substituting here though seems right to me based on a cursory reading of the cargo make docs

Comment (topecongiro) [reply Makefile.toml:38]: Looks like we need to change `script_runner` on windows: https://github.com/sagiegurari/cargo-make/issues/62.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -39,13 +39,15 @@ jobs:
         override: true
         profile: minimal
         default: true
+
+    - name: cargo-make
+      run: cargo install --force cargo-make
```
[the remark this answers is not in the document]
Comment (topecongiro) [reply .github/workflows/linux.yml]: Thanks for the advice! I have added rustc-cargo-make action.

[the remark this answers is not in the document]
Comment (topecongiro) [reply .github/workflows/linux.yml:44]: Hmm, unfortunately, it looks like the rustc-cargo-make does not pick up the latest cargo-make even :thinking: I will just use `cargo install` approach for now.

**Comment by topecongiro** [comment]
Great, finally it's green :pleading_face:

**Pull request merged**